src/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubAdapterReceiver.cs

```diff
@@ -63,25 +64,25 @@ public int GetMaxAddCount()
         }
 
         public EventHubAdapterReceiver(EventHubPartitionSettings settings,
-            Func<string, IStreamQueueCheckpointer<string>, Logger, ITelemetryProducer, IEventHubQueueCache> cacheFactory,
+            Func<string, IStreamQueueCheckpointer<string>, ILoggerFactory, ITelemetryProducer, IEventHubQueueCache> cacheFactory,
             Func<string, Task<IStreamQueueCheckpointer<string>>> checkpointerFactory,
-            Logger baseLogger,
+            ILoggerFactory loggerFactory,
             IQueueAdapterReceiverMonitor monitor,
             Factory<NodeConfiguration> getNodeConfig,
             ITelemetryProducer telemetryProducer,
-            Func<EventHubPartitionSettings, string, Logger, ITelemetryProducer, Task<IEventHubReceiver>> eventHubReceiverFactory = null)
+            Func<EventHubPartitionSettings, string, ILogger, ITelemetryProducer, Task<IEventHubReceiver>> eventHubReceiverFactory = null)
         {
             if (settings == null) throw new ArgumentNullException(nameof(settings));
             if (cacheFactory == null) throw new ArgumentNullException(nameof(cacheFactory));
             if (checkpointerFactory == null) throw new ArgumentNullException(nameof(checkpointerFactory));
-            if (baseLogger == null) throw new ArgumentNullException(nameof(baseLogger));
+            if (loggerFactory == null) throw new ArgumentNullException(nameof(loggerFactory));
             if (monitor == null) throw new ArgumentNullException(nameof(monitor));
             if (telemetryProducer == null) throw new ArgumentNullException(nameof(telemetryProducer));
             this.settings = settings;
             this.cacheFactory = cacheFactory;
             this.checkpointerFactory = checkpointerFactory;
-            this.baseLogger = baseLogger;
-            this.logger = baseLogger.GetSubLogger("receiver", "-");
+            this.loggerFactory = loggerFactory;
+            this.logger = this.loggerFactory.CreateLogger($"{this.GetType().FullName}.{settings.Partition}");
```

Comment:
Should probably use hub path and partition.  Partitions alone won't be sufficient if multiple EHSP are configured.

Reply:
hub name or provider name.. either one.